### PR TITLE
Added documentation for System.Diagnostics APIs targeted for 3.0

### PR DESCRIPTION
**Pull request by jozkee**
APIs affected:
* Activity:
  * SetIdFormat(System.Diagnostics.ActivityIdFormat)
  * SetParentId(System.Diagnostics.ActivityTraceId,System.Diagnostics.ActivitySpanId,System.Diagnostics.ActivityTraceFlags)
* DiagnosticListener
  * Subscribe(System.IObserver{System.Collections.Generic.KeyValuePair{System.String,System.Object}},System.Func{System.String,System.Object,System.Object,System.Boolean},System.Action{System.Diagnostics.Activity,System.Object},System.Action{System.Diagnostics.Activity,System.Object})

### CODE REVIEW

**Comment by Jozkee** [comment]
@mairaw @rpetrusha @carlossanlop 
Authors: @lmolkova @vancem
Area Owner: @sywhang

**Comment by carlossanlop** [comment]
@lmolkova for some reason we can't add you to the list of reviewers (you don't show up). Can you please take a look?

File: xml/System.Diagnostics/DiagnosticListener.xml [hunk 1]
```diff
@@ -439,10 +439,10 @@ If `isEnabled` is `null`, no filtering is done (all overloads of <xref:System.Di
         <Parameter Name="onActivityExport" Type="System.Action&lt;System.Diagnostics.Activity,System.Object&gt;" Index="3" FrameworkAlternate="netcore-3.0" />
       </Parameters>
       <Docs>
-        <param name="observer">To be added.</param>
-        <param name="isEnabled">To be added.</param>
-        <param name="onActivityImport">To be added.</param>
-        <param name="onActivityExport">To be added.</param>
+        <param name="observer">A subscriber.</param>
+        <param name="isEnabled">A delegate that filters events based on their name and up to two context objects (which can be null), or null to if an event filter is not desirable.</param>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/DiagnosticListener.xml]: Is this talking about a C# null? If yes, then it should it be `<see langword="null" />`. If not (like with Json Nulls), then ignore my suggestion:

```suggestion
        <param name="isEnabled">A delegate that filters events based on their name and up to two context objects (which can be <see langword="null" />), or <see langword="null" />to if an event filter is not desirable.</param>
```

File: xml/System.Diagnostics/Activity.xml [hunk 2]
```diff
@@ -604,10 +604,18 @@ The root ID is a substring from the <xref:System.Diagnostics.Activity.Id> or the
...
+        <summary>Sets the ID format on this <see cref="T:System.Diagnostics.Activity" /> before it is started.</summary>
+        <returns><see langword="this" /> for convenient chaining.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+
+## Remarks
+
+The specified format will take precedence over `Parent.IdFormat`, `ParentId` format, `DefaultIdFormat` and `ForceDefaultIdFormat`.
+
+]]></format>
```
Comment (carlossanlop) [inline xml/System.Diagnostics/Activity.xml]: nit: Alignment:

```suggestion
          ]]></format>
```

**Comment by lmolkova** [comment]
LGTM, thanks!

File: xml/System.Diagnostics/Activity.xml [hunk 3]
```diff
@@ -604,10 +604,18 @@ The root ID is a substring from the <xref:System.Diagnostics.Activity.Id> or the
...
-        <remarks>To be added.</remarks>
+        <param name="format">One of the enumeration values that specifies the format of the <see cref="P:System.Diagnostics.Activity.Id" /> property.</param>
+        <summary>Sets the ID format on this <see cref="T:System.Diagnostics.Activity" /> before it is started.</summary>
+        <returns><see langword="this" /> for convenient chaining.</returns>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+
+## Remarks
+
+The specified format will take precedence over `Parent.IdFormat`, `ParentId` format, `DefaultIdFormat` and `ForceDefaultIdFormat`.
```
Comment (rpetrusha) [inline xml/System.Diagnostics/Activity.xml]:
```suggestion
The specified format will take precedence over `Parent.IdFormat`, the <xref::System.Diagnostics.Activity.ParentId> format, <xref::System.Diagnostics.Activity.DefaultdFormat>, and <xref::System.Diagnostics.Activity.ForceDefaultIdFormat>.
```
except for Parent.IdFormat, I thought that these are all better as links.

**Comment by Jozkee** [comment]
Done. This should be ready to merge, then.

**Comment by rpetrusha** [comment]
It's ready to merge, except that, of course, there are build errors. I've fixed them, so the PR can be merged once the build completes successfully.

**Comment by mairaw** [comment]
There was one more left.

**Comment by Jozkee** [comment]
> There was one more left.

@mairaw thanks for catching that and thanks for merging.

**Pull request merged**